### DISCUSSION TITLE
Is PyPSA allowed to run FBMC models?

### DISCUSSION THREAD

**Discussion by Bitzilla-no** [PyPSA/PyPSA#581] [Q&A]
Hello, I am Carlos Tellez, a master's candidate in electrical engineering working on my thesis at the University of South Eastern Norway. My supervisor and I are thinking of developing an application to calculate and compare the NTC (Net Transfer Capacity) and FBMC (Flow-Based Market Coupling) market models for the Nordic countries.

I have been reviewing the PyPSA code, but I can't find a function that allows me to model and optimize the network for FBMC. Therefore, I would like to ask the PyPSA community: is it possible to model electricity markets based on FBMC using PyPSA? If so, how can I do it? Any guidance or advice would be greatly appreciated.

Thank you in advance for your help!

**Comment by Cellophil** [comment]
Hi @Bitzilla-no ,

We are doing that with pypsa, but it's a bit of custom code here and there. I believe there's a jupyter in the tutorials on that topic.
So pypsa can calculate the ptdf matrix for you (be aware that you need to attach the right axis, ie buses_o and lines_i...). If you need N-1 you need to iterate. Then, think about the GSK to apply for each biddingzone and aggregate over the buses. Viola the constraints are the rows of the ptdf matrix.
RAM calculation and 70% rule can be done by presolving and reading off the line load. Good luck, it's a big topic!

Best,
Andreas
